google/cloud/spanner_v1/snapshot.py

```diff
@@ -92,6 +95,7 @@ def _restart_on_unavailable(
     ):
         iterator = method(request=request)
     while True:
+        attempt += 1
```

Comment:
I think that this should be moved into the `except ServiceUnavailable` block. The logic in this function is:
1. Iterate until you encounter a `resume_token`. When we see that, then yield what is in the buffer, as we know that it will be safe to retry.
2. If a `ServiceUnavailable` error is encountered, then do another call (a retry) and replace the underlying iterator with the one that is returned by that new call.

Reply:
It should also be incremented in case of an `InternalServerError` that is retryable.

Reply:
Or can it be removed altogether? It does not seem to be used?

Reply:
It was being used as a side effect but sure I've removed it and performing the increment appropriately where directly used.